### ISSUE TITLE
Feature request: UX updates to theme customizer

### ISSUE THREAD

**Issue by manda-ikaink** [fusioncms/fusioncms#567]
### Describe the solution you'd like
There are a couple of small UX updates that would be nice to have on the theme customizer:

- Have an alert pop up before you leave the customizer page. Multiple times I've accidentally clicked something and lost all my unsaved changes.
- When saving customized options, I find myself unsure if clicking the save button did anything at all as it does not seem very responsive to the interaction. A toast notification at the bottom right of the screen will pop up eventually. But this notification is delayed, showing up only once the save is complete, and is very easy to over look in it's current position. Clicking the save button and having some sort of loading state on the button or window so it's clear the save event is taking place would be suggested.

### Feature request checklist
- [x] This is a single feature (i.e. not a re-write of the entire CMS)
- [x] The title starts with "Feature request: " and is followed by a clear feature name (Ex: `Feature request: add konami code to control panel`)
- [x] I have [searched for existing issues](https://github.com/fusioncms/fusioncms/issues) and to the best of my knowledge this is not a duplicate
